mpd/__init__.py

```diff
@@ -21,120 +22,381 @@
+            if warn:
+                self.loggercmd("update_statistic while not connected",'e')
+            return
+        if (len(self._statistic_items) <= 0):
+            if warn:
+                self.loggercmd("statistic: no items to refresh",'e')
+            return
+        self.loggercmd("requesting statistic",'d')
+        stats = self._send('stats')
+        self.refreshItems(self._statistic_items,stats,warn)
 
-    def _play_file(self, item, caller=None, source=None, dest=None):
-        if caller != 'MPD':
-            if item.conf['mpd_file'] == 'value':
-                self.play_file(item())
+    def refreshItems(self,subscribedItems,response,warn):
+        #1. check response for the internal items and refresh them
+        if 'state' in response:
+            val = response['state']
+            if val == 'play':
+                self._internal_tems['isPlaying'] = True
+                self._internal_tems['isPaused'] = False
+                self._internal_tems['isStopped'] = False
+            elif val == 'pause':
+                self._internal_tems['isPlaying'] = False
+                self._internal_tems['isPaused'] = True
+                self._internal_tems['isStopped'] = False
+            elif val == 'stop':
+                self._internal_tems['isPlaying'] = False
+                self._internal_tems['isPaused'] = False
+                self._internal_tems['isStopped'] = True
             else:
-                self.play_file(item.conf['mpd_file'])
+                self.loggercmd("unknown state: {}".format(val),'e')
+        if 'volume' in response:
+            val = float(response['volume'])
+            if val <= 0:
+                self._internal_tems['isMuted'] = True
+            else:
+                self._internal_tems['isMuted'] = False
+                self._internal_tems['lastVolume'] = val
+        if 'Name' in response:
+            val = response['Name']
+            if val:
+                self._internal_tems['currentName'] = val
+        #2. check response for subscribed items and refresh them
+        for key in subscribedItems:
+            #update subscribed items (if value has changed) which exist directly in the response from MPD
+            if key in response:
+                val = response[key]
+                item = subscribedItems[key]
+                if item.type() == 'num':
+                    try:
+                        val = float(val)
+                    except:
+                        self.loggercmd("can't parse {} to float".format(val),'e')
+                        continue
+                if item() != val:
+                    self.loggercmd("update item {} with {}".format(key,val),'d')
+                    self.setItemValue(item,val)
+            #update subscribed items which do not exist in the response from MPD
+            elif key == 'playpause':
+                item = subscribedItems[key]
+                self.setItemValue(item,self._internal_tems['isPlaying'])
```

Comment:
I get the same problem as with `volume` here (item `playpause` is update in every cycle). How about checking the same way as in the lines above (219 ff):

            elif key == 'playpause':
                item = subscribedItems[key]
                val = self._internal_tems['isPlaying']
                if item() != val:
                    self.setItemValue(item,val)

Reply:
Update: I changed that to a PR into your repo. Please try if it get's included into the PR to the main repo if you merge my Sub-PR. Just curious :)